### ISSUE TITLE
Support babel-preset-env

### ISSUE THREAD

**Issue by matthewp** [stealjs/steal#1277]
__tldr;__ give some special capabilities to babel-preset-env, so that it can do its thing.

> This was discussed on a [recent live stream (7:29)](https://youtu.be/ndAlLvRmyNI?t=7m29s), a [previous live stream (40:06)](https://youtu.be/ieZJrPwSZYQ?t=40m06s), and a [contributors meeting (1:11:36)](https://youtu.be/95OX6-qii24?t=1h11m36s).

<img width="645" alt="screen shot 2017-09-22 at 10 05 52 am" src="https://user-images.githubusercontent.com/361671/30748188-826bf426-9f7d-11e7-854b-edaa98f9efe6.png">

# The Problem

In Babel 5 instead of specifying plugins you wanted to use, it would just transpile everything to ES5. This make it easy to use, but difficult to customize. What if you wanted to target browsers that support classes? It would still compile to the more verbose (and some times incompatible) ES5 form.

In Babel 6+ you have to specify which plugins to use. So if you want to transpile classes you have to add that plugin (or a preset, which is a collection of plugins).

This makes Babel much more flexible, but sacrifices some of that ease-of-use that Babel 5 had.

The solution that the babel team came up with is babel-preset-env. It works similar to Babel 5, but allows you to say which browsers you are trying to target, and it will know which plugins are necessary.

This is great! However, steal can't support babel-preset-env, because it needs to dynamically `require` some modules at run time.

# The Solution

The good knows is that it only tries to `require` other Babel plugins, and all of these are available in the babel-standalone package that steal has. So all of these things *are* available, it's just that Steal throws when a module tries to require something that it doesn't have in its dependency graph.

So __the solution__ is to give some special permission to babel-preset-env, so that it __can__ dynamically require those plugins.

# Implementation

I think this won't be too difficult to implement. I think we simply add a flag for babel-preset-env via meta configuration, that says that it can try and dynamically import modules not part of its dependency graph. 

When the `require()` calls happen, we'll know to look into the babel-standalone register and get the modules from there.

# Tasks

* [ ] - Meta configuration allowing dynamic requires.
* [ ] - Piping those requires over to the babel-standalone registry.
